### PR TITLE
Report snap modal

### LINKED ISSUE #1556: Implement frontend for report snap form modal

### PR DESCRIPTION
**Pull request by bartaz**
Fixes #1556

Adds a modal dialog to report snap on snap details page.
Data is posted to Google Form.

### QA
- ./run or https://snapcraft-io-canonical-websites-pr-1568.run.demo.haus/
- go to snap details page of any suspicious snap
- scroll to the bottom of the page, click on 'Report this app' link
- fill the details in the form, Submit
- success message should be shown
- data should be added to [Report snap Google Sheet](https://docs.google.com/spreadsheets/d/1ju2IKRiMsVK17b6Fmtj335sj3xaawXcN4hYGZvxwR6s/edit)

screenshot 2019-02-06 at 16 20 42

### CODE REVIEW

**Comment by therealjuan** [comment]
Amazing! Works like a charm... is there any way we can have the modal closed when pressing `Esc` or clicking anywhere out of the dialog?

**Comment by bartaz** [comment]
@therealjuan Yep, I was thinking about it as well. Will add it.

**Comment by therealjuan** [comment]
Sorry to be nick-picking but can we have a loading button after you press submit report? The demo service is slow and it takes some time to process so some indication to users that the request is being processed might be handy

**Comment by bartaz** [comment]
@therealjuan Not sure if it's demo service or Google Sheets thingy that is making it slow. But yes, some loading indicator would be useful.

**Comment by therealjuan** [comment]
👍 🎉 :shipit:  👍 🎉 :shipit:  LGTM 👍 🎉 :shipit:  👍 🎉 :shipit:  👍 🎉 :shipit:

File: static/js/public/snap-details/reportSnap.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
+import "whatwg-fetch";
+
+function toggleModal(modal, show) {
+  if (typeof show === "undefined") {
+    show = modal.style.display === "none";
```
Comment (Lukewh) [inline static/js/public/snap-details/reportSnap.js]: we should use `.u-hide` for toggling visibility

File: static/js/public/snap-details/reportSnap.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  }
+
+  if (show) {
+    showForm(modal);
+    modal.style.display = "";
```
Comment (Lukewh) [inline static/js/public/snap-details/reportSnap.js]: same again

File: static/js/public/snap-details/reportSnap.js (hunk 1 above)
Comment (bartaz) [reply static/js/public/snap-details/reportSnap.js]: Forgot completely about this utility. Thanks.

**Pull request merged**